ansible_collections/f5networks/f5_modules/plugins/modules/bigip_device_group.py

```diff
@@ -575,6 +626,10 @@ def __init__(self):
                 type='bool',
                 default='no'
             ),
+            asm_sync=dict(
```

Comment:
no defaults , if you want to have create default add it to create stage, also if we created device group before without this option specified then having default for create is pointless, we only add default create arguments for mandatory options that are required for new resources

Reply:
Again I used `auto_sync` as a working example. Removing this would not restore the BIG-IP default behaviour of not syncing ASM policies via the device-group. For example when `asm_sync: yes` is just removed from the ansible task this setting would not be updated and `asmSync` would stay `enabled` in the API.
The above code mirrors the `auto_sync` behaviour exactly, which happens to be disabled by default as well.

Reply:
if the user has this changed to enabled already, then running playbook without this parameter will cause the change, remove it

Reply:
we do not use defaults for api parameters only for parameters that are not linked directly to API

Reply:
and to be honest we should change it here cause its not very consistent with rest of code

Reply:
I see, you are referring to existing configuration conflicting with the defaults in this module. I would still keep the default because the behaviour for `asm_sync` would exactly match the existing behaviour of `auto_sync`. I would also assume that any configuration object created via or added to ansible is _owned_ by ansible and MUST NOT be modified outside ansible.

For example assume the below device group was created outside of ansible:
```hcl
cm device-group my-device-group {
    auto-sync enabled
}
```

Then running the below ansible task:
```yaml
- name: create device-group
  f5networks.f5_modules.bigip_device_group:
    name: 'my-device-group'
  delegate_to: localhost
```
Will result in:
```sh
changed: [bigip -> localhost({{ provider.server }})] => {"auto_sync": "no", "changed": true}
```
```hcl
cm device-group my-device-group { }
```

The same is true for `asm_sync` but it just mirrors existing behaviour. If the existing behaviour is undesired I would rather see this as part for a bigger refactoring of this module than adding new behaviour through this feature addition.